### DISCUSSION TITLE
What does the exclemation mark left of the dB indicate?

### DISCUSSION THREAD

**Discussion by codingchipmunk** [wwmm/easyeffects#2832] [Q&A] [answered]
Hello everyone,
I have always had issues with equalizers and PulseAudio, where increasing the loudness of frequencies led to distortion / scratching. The same happened when using the EQ from easy effects, only that - thanks to the status bar - I now know that this only happens when dB get positive. Also, easy effects shows an exclamation mark besides the dB when this happens.

So, independent of my audio issue, I would like to ask if the exclamation mark pops up if there is an actual issue with PulseAudio and / or its configuration or just shows up when dB are greater than 0 (unfortunately there isn't any tooltip on mouseover).

![Screenshot_429](https://github.com/wwmm/easyeffects/assets/36039591/596c3f33-7762-405d-9d44-dec7a6004512)

**Comment by wwmm** [comment, marked as answer]
> or just shows up when dB are greater than 0

That is it. Before sending the signal to the sound card EasyEffects measures its level. This is what is shown in its bottom bar. If the level is above `0 dB` the warning in the image is shown so the user can know the signal level is in a dangerous zone. It is in general fine to have the level above 0 dB in the other plugins level meters. But you do not want the output level in the bottom bar to get this high because this what your soundcard is going to get.

**Reply by wwmm** [reply]
Based on your image you are definitely applying too much amplification. That is not safe for your soundcard or your health. In most cases it is better to do what is called "subtractive equalization". Instead of amplifying the frequencies you want to be boosted reduce the level of the ones that you want to be attenuated. In you image you would keep the low frequencies gain close to "0 dB" and use a negative gain for the others you kept at zero gain.

**Reply by wwmm** [reply]
If the output audio level is too low you can always use the `output gain` slide at the bottom to compensate the reduction. It is better than applying such a large gain to each frequency band.

**Reply by wwmm** [reply]
And if all you want is too boost the bass a little it may be easier to just use the `bass enhancer` plugin instead of the equalizer that is harder to set properly.

**Reply by fxzzi** [reply]
partially related, is there any way to see an estimated peak gain like is possible in EqualizerAPO on the bottom left? I'd like to prevent peaking with a few filters active, and want to get peak gain at 0dB

**Reply by wwmm** [reply]
>is there any way to see an estimated peak gain

Peak gain of what exactly? We show at the bottom left the peak at the end of the pipeline. For individual plugins output level it is necessary to look at each plugin window.

**Reply by fxzzi** [reply]
As equalizerAPO shows, it makes a calculation / estimate of the maximum gain your setup can reach, allowing you to set your preamp lower to account for it.

**Reply by wwmm** [reply]
> As equalizerAPO shows, it makes a calculation / estimate of the maximum gain your setup can reach, allowing you to set your preamp lower to account for it.

Oh... I see. That can be a little complicated to do accurately. First because we handle many different plugins and the user is free to put more than one instance of each of them in the pipeline. Second because most of the plugins we use are third party plugins whose code and algorithm I am not familiar with. In APO the equalizer code is part of APO and not an external plugin. So they know the algorithm and can estimate things like these based on how the controls are set. Here I will have to find a mathematical model entirely through trial and error hoping its results would be good.

As this is not going to be done I suggest an alternative approach. Put the `limiter` plugin at the end of the pipeline and a `level meter` plugin right before it.  Then use EasyEffects for some minutes with your usual workflow. The Level meter plugin will record the highest peak that went through it  in the time window it was running. It is not the same as what APO is doing but it will give you an idea of how high the peak is getting. And with the limiter at the end the will be some protection in case something unexpected happens.
